protobuf/options.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package solo.io.protocgenopenapi;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = "github.com/solo-io/protoc-gen-openapi/protobuf/options";
+
+// FieldOptions expose custom configuration at the field level.
+// The schema generator parses these options from the proto descriptors to generate appropriate OpenAPI schemas.
+message FieldOptions {
+    // Enable this option to treat this field as an unstructured object in the OpenAPI schema.
+    // This can be set to disable infinite recursion of proto fields.
+    optional bool type_object = 1;
```

Comment:
Do we want to maintain the same API as we used to have? `disable_openapi_validation`?

Reply:
No, it was pretty ambiguous `disable_openapi_validation` vs `disable_openapi_type_validation`. I'd rather be explicit here with regards to proto->openapi mappings.